avocado/core/loader.py

```diff
@@ -803,6 +809,19 @@ def _get_methods_info(statement_body, class_tags):
 
         return methods_info
 
+    def _find_python_unittests(self, test_path):
+        result = []
+        class_methods = safeloader.find_class_and_methods(test_path,
+                                                          _RE_UNIT_TEST)
+        for klass, methods in class_methods.iteritems():
+            if test_path.endswith(".py"):
+                test_path = test_path[:-3]
```

Comment:
You care only with `.py`? Ok, safeloader cannot parse `.pyc`, but the point is that what you really want is to remove the extension, whatever it is. I'd use `os.path.splitext()` for extra flexibility.

Reply:
`python -m unittest` only discovers files ending with `.py?` and we are only able to parse the source `.py` files. With that said I actually think we should move the `if endswith` to early stage and skip files that do not end with `.py`.